client/my-sites/plan-features-2023-grid/billing-timeframe.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+	isWpComFreePlan,
+	isWpcomEnterpriseGridPlan,
+	PLAN_BIENNIAL_PERIOD,
+	PLAN_ANNUAL_PERIOD,
+} from '@automattic/calypso-products';
+import { getCurrencyObject } from '@automattic/format-currency';
+import { localize, TranslateResult, useTranslate } from 'i18n-calypso';
+import { FunctionComponent } from 'react';
+import { useSelector } from 'react-redux';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+
+interface Props {
+	planName: string;
+	billingTimeframe: TranslateResult;
+	billingPeriod: number | null | undefined;
+	rawPrice: number | null;
+	maybeDiscountedFullTermPrice: number | null;
+	annualPricePerMonth: number | null;
+	isMonthlyPlan: boolean;
+}
+
+function usePerMonthDescription( {
+	rawPrice,
+	maybeDiscountedFullTermPrice,
+	annualPricePerMonth,
+	isMonthlyPlan,
+	planName,
+	billingPeriod,
+}: Omit< Props, 'billingTimeframe' > ) {
+	const translate = useTranslate();
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+
+	if ( isWpComFreePlan( planName ) || isWpcomEnterpriseGridPlan( planName ) ) {
+		return null;
+	}
+
+	if ( isMonthlyPlan ) {
+		if ( null !== rawPrice && null !== annualPricePerMonth ) {
+			if ( annualPricePerMonth < rawPrice ) {
+				return translate( `Save %(discountRate)s%% by paying annually`, {
+					args: {
+						discountRate: Math.round( ( 100 * ( rawPrice - annualPricePerMonth ) ) / rawPrice ),
+					},
+				} );
+			}
+		}
+	}
+
+	if ( ! isMonthlyPlan ) {
+		const fullTermPriceObj =
+			currencyCode && null !== maybeDiscountedFullTermPrice
+				? getCurrencyObject( maybeDiscountedFullTermPrice, currencyCode )
+				: null;
+		const fullTermPriceText = `${ fullTermPriceObj?.symbol }${ fullTermPriceObj?.integer }`;
```

Comment:
I know this is just moving an existing file but I thought I'd mention it:

Discussed in https://github.com/Automattic/wp-calypso/issues/74605, if you don't need to add HTML to the price, you should use `formatCurrency()` instead of trying to format the price yourself. Here, for example, you are not showing any decimals for the price even if it has decimals, which could be misleading for users (maybe that was intentional but I don't know it's a great idea).

Reply:
Thanks @sirbrillig . Also for adding an issue on your board already for it https://github.com/Automattic/wp-calypso/issues/74605. Is someone from @Automattic/shilling going to pick this up?

p.s. the relevant code above was here due to a change of base branch. PR's rebased now, hence no longer visible

Reply:
@jjchrisdiehl is giving it a try in https://github.com/Automattic/wp-calypso/pull/74683. We'll keep you in the loop.

Reply:
@chriskmnds I'm not sure how we could coordinate the changes found [here](https://github.com/Automattic/wp-calypso/issues/74605#issuecomment-1478835701)

I can merge them first, then rebase them into your current work, or you can manually add them in since the changes aren't that involved. Let me know what you think!

Reply:
Thanks, @jjchrisdiehl for being considerate. Feel free to merge your changes. The related files were from before we rebased here, so should be ok. but in any case, we can handle conflicts after. This PR may not ship in its current form either